ci/docker/base.test.Dockerfile

```diff
@@ -33,6 +34,7 @@ RUN apt-get install -y -qq \
     sudo unzip unrar apt-utils dialog tzdata wget rsync \
     language-pack-en tmux cmake gdb vim htop \
     libgtk2.0-dev zlib1g-dev libgl1-mesa-dev \
+    liblz4-dev libunwind-dev musl-dev libncurses5 \
```

Comment:
Could switching to musl as the libc cause problems here? What is this docker file used for, are any artifacts built in it? If no, it is probably fine :)

Reply:
The Dockerfile is for the images the CI runners run their tests in - so yeah, the wheels are built within it.

These dependencies are required to install memray, but we may be able to omit musl. I'll give this another try

Reply:
Thanks! If we can get rid of musl, that would be perfect! If we need it, we need to make sure the binaries / shared libraries are not linked against it :)

Reply:
It works without musl - so should be good to go!